Tools/CISupport/ews-build/steps.py

```diff
@@ -3323,6 +3323,11 @@ def doStepIf(self, step):
                     (self.getProperty('fast_commit_queue') or self.getProperty('passed_mac_wk2')))
 
     def setLayoutTestCommand(self):
+        classification = self.getProperty('classification', [])
+        if not isinstance(classification, list):
+            classification = []
+        is_initial_wpt_upload = self.getProperty('action', '') == 'opened' and 'Test-import' in classification
```

Comment:
Ideally, we'd run all newly added tests regardless of where they came from. We already have the logic to identify those for the stress test.

Reply:
WPT imports aren't just adding tests, though. They're modifying existing tests as well. And we want baselines on all configurations, not just the configuration covered by stress tests.

Reply:
> WPT imports aren't just adding tests, though. They're modifying existing tests as well.

This is true of how stress test behaves, too.

> And we want baselines on all configurations, not just the configuration covered by stress tests.

I am not saying that people should rely on stress tests. I am saying that instead of special casing WPT imports, this change should fix the problem for all PRs importing or updating many tests. And that we already have most of the necessary code written, because we needed it for stress tests.

Reply:
My understanding is that our problem with WPT imports is about the "importing" piece, not about mass test modification. Basically, during a WPT import, the importer hasn't run the tests they're importing and they know that the baselines either don't exist or are wrong. So we need to disable our failure limit so that we get a result for every test, at least on the initial upload. After that, EWS should function as it normally does.

I can't think of a circumstance, other than WPT import, where we want EWS to disable the failure limit. In other cases, I would expect engineers to have specifically run the tests they're modifying, I don't think disabling the failure limit (and retries) would be desirable.

Reply:
We had many requests for the same behavior outside of WPT imports. Not all of them would be solved by looking at changed tests (e.g. render tree rebases when new information is added to output), but I'm pretty certain that WPT isn't the only case.

Reply:
Do we think this behavior is common enough to justify a `no-failure-limits` label? That is, if I recall the alternate proposal. We would basically have a `no-failure-limits` label that `git-webkit pr` applies to initial WPT uploads by default, but also, any PR could add that label to opt in to the behavior.

Reply:
My alternative proposal is to look at changed tests, but `no-failure-limits` seems reasonable too.

Reply:
I still don't understand your alternative proposal focusing on changed tests. PRs modify layout tests all the time, and we very rarely want to disable retries and remove the failure limit. Even for WPT uploads, that's only something we want to do on the initial upload of a PR. After the baselines have been gathered, we want EWS to return to its normal behavior.

The only way I can see to generalize the underlying goal is to increase the failure limit based on the number of modified tests. That would let WPT uploads get baselines for all modified tests. It also runs the risk of wasting machine time, though, because we couldn't generally disable retries for all changes which modify layout tests.

Reply:
> The only way I can see to generalize the underlying goal is to increase the failure limit based on the number of modified tests.

Yes, this is what I am talking about.

Reply:
I will note we don't necessarily have an easy way to know what the total modified tests are: if someone modifies testharness.js practically we need to re-run thousands of tests, even if only one file has changed. If we make the default vary based on number of modified tests, we probably still need as escape-hatch both for WPT imports and other cases.